test/test_create_runtime_policy.py

```diff
@@ -865,19 +881,31 @@ def test_digest_algorithm_priority_exceptions(self):
             expected_algo = case["expected_algo"]
             expected_source = case["expected_source"]
 
-            with self.assertLogs("policy.create_runtime_policy", level="DEBUG") as logs:
+            try:
+                # Since the logger does not propagate, we cannot use
+                # something as simple as self.assertLogs here.
+                # Let's replace the log stream so that we can capture it.
+                stderr = StringIO()
+                self.logger.setStream(stderr)
+
+                # Now let's run it with the arguments and capture stderr.
                 _policy = create_runtime_policy.create_runtime_policy(args)
+                logs = stderr.getvalue()
+            finally:
+                # Restore the stream.
+                self.logger.setStream(sys.stderr)
```

Comment:
@sergio-correia Do you think it would be possible to make this a private method so that we can reuse it without duplicating the code?

Not a blocker, just to make it a bit nicer.

Reply:
Please take another look, now it should be better.